src/com/lftechnology/java/training/srijan/basics/coreapp/controller/EmployeeManagementController.java

```diff
@@ -0,0 +1,307 @@
+
+package com.lftechnology.java.training.srijan.basics.coreapp.controller;
+
+import com.lftechnology.java.training.srijan.basics.coreapp.domain.Employee;
+import com.lftechnology.java.training.srijan.basics.coreapp.domain.SearchEmployee;
+import com.lftechnology.java.training.srijan.basics.coreapp.domain.UserRole;
+import com.lftechnology.java.training.srijan.basics.coreapp.service.EmployeeServiceImpl;
+import com.lftechnology.java.training.srijan.basics.coreapp.service.ServiceFactory;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.List;
+import java.util.Scanner;
+
+public class EmployeeManagementController {
+
+	private static final Logger LOGGER =
+		Logger.getLogger(EmployeeManagementController.class.getName());
+	List<Employee> employees;
+	Employee employee;
+	Employee e = new Employee();
+	EmployeeServiceImpl employeeServiceImpl =
+		ServiceFactory.getEmployeeService();
+	private int choiceUserRole;
+	private int userStatus;
+	private String username;
+	private String fullname;
+	private String department;
+	private String address;
+	private String password;
+
+	public EmployeeManagementController() {
+		this.choiceUserRole = 0;
+		this.userStatus = 0;
+		this.username = "";
+		this.fullname = "";
+		this.department = "";
+		this.address = "";
+		this.password = "";
+	}
+
+	/**
+	 * <p>Sets username and password</p>
+	 * 
+	 * @param userName
+	 * @param password
+	 * @return
+	 * @author srijan
+	 */
+	public UserRole save(String userName, String password) {
+
+		e.setUserName(userName);
+		e.setPassword(password);
+		return employeeServiceImpl.save(e);
+	}
+
+	/**
+	 * <p>Calls different methods depending upon user choice</p>
+	 * 
+	 * @param adminChoice
+	 * @author srijan
+	 */
+	public void adminUserOptionHandler(int adminChoice) {
+
+		switch (adminChoice) {
+		case 1:
+			Employee employeeDetails = inputUserInfo(e);
+			employeeServiceImpl.addEmployee(employeeDetails);
+			break;
+		case 2:
+			SearchEmployee emp = new SearchEmployee();
+			LOGGER.log(
+				Level.INFO,
+				"Search By:\n1.Department\n2.Fullname\n3.Address\n4.Default");
+			emp = choiceOfSearch();
+			if ("default".equals(emp.getSearchChoice())) {
+				employees = employeeServiceImpl.viewAllUser(e);
+				LOGGER.log(Level.INFO, "{0}", employees);
+			}
+			else {
+				e = employeeServiceImpl.search(emp);
+				LOGGER.log(Level.INFO, "{0}", e);
+			}
+			break;
+		case 3:
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this switch case number of lines from 7 to at most 5, for example by extracting code into methods. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1151)